src/main/scala/bitboard/Board.scala

```diff
@@ -40,7 +40,10 @@ case class Board(
     else None
 
   def pieceAt(s: Pos): Option[Piece] =
-    (colorAt(s), roleAt(s)).mapN(Piece.apply)
+    for
```

Comment:
So, you prefer for comprehension then cats `mapN`?

Reply:
on sundays and wednesdays, yes.

Here it doesn't add much eye-candy imo, and it's almost certainly slower than plain `for` or `flatMap`